### PR TITLE
Implementation of 3R+ schemes from Carpenter Kennedy Lewis(2000)

### PR DESCRIPTION
**Pull request by deeepeshthakur**
@ChrisRackauckas @ranocha Implemented LowStorageRK methods from [Kennedy, Carpenter, Lewis (2000)](https://www.sciencedirect.com/science/article/pii/S0168927499001415). `[author's name]LDDRK[stage][order]_[embedded_order for error estimation]_3R`.
A few tests are marked as `broken`, because even lower `dts` were giving too much precision.
Memory usage ratio is 10 for `adaptive` versions and 9 for `non-adaptive`.
**Methods Implemented:**
1. `CKLLSRK54_3C_3R`
2. `CKLLSRK54_3M_3R`
3. `CKLLSRK54_3N_3R`
4. `CKLLSRK85_4C_3R`
5. `CKLLSRK85_4M_3R`
6. `CKLLSRK85_4P_3R`

#493

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
>A few tests are marked as `broken`, because even lower `dts` were giving too much precision.

Then change the dts or use bigfloats?

**Comment by deeepeshthakur** [comment]
Will do. It's one of the tests for `CKLLSRK85_4M_3R`.

**Comment by deeepeshthakur** [comment]
@ChrisRackauckas 
This is essentially the test I've marked broken. It passes the tests when changed it to `BigFloats`. And also the plot is linear.
```
f = (u,p,t)->cos(t)
(::typeof(f))(::Type{Val{:analytic}},u0,p,t) = sin(t)
prob_ode_sin = ODEProblem(f, BigFloat(0.0),(BigFloat(0.0),BigFloat(1.0)))

dts = BigFloat(2) .^(-8:1:-4)
sim = test_convergence(dts, prob_ode_sin,CKLLSRK85_4M_3R())
@test sim.𝒪est[:final] ≈ OrdinaryDiffEq.alg_order(alg) atol=testTol 
plot(sim)
```
image

This was the plot when using `Float64`
```
f = (u,p,t)->cos(t)
(::typeof(f))(::Type{Val{:analytic}},u0,p,t) = sin(t)
prob_ode_sin = ODEProblem(f, 0.0,(0.0,1.0))

dts = 2.0 .^(-8:1:-4)
sim = test_convergence(dts, prob_ode_sin,CKLLSRK85_4M_3R())
plot(sim)
```
image

But actually changing `uType` to BigFloat in test problems will change them for all the tests, because they are common to all the methods.

**Comment by ranocha** [comment]
You can also change `dts` for `test_problems_only_time`. For me, the following tests pass:
```julia
alg = CKLLSRK85_4M_3R()
dts = 1.6 ./ 2 .^(6:-1:2)
for prob in test_problems_only_time
  sim = test_convergence(dts, prob, alg)
  @test sim.𝒪est[:final] ≈ OrdinaryDiffEq.alg_order(alg) atol=testTol
end
dts = 1 ./ 2 .^(6:-1:2)
for prob in test_problems_linear
  sim = test_convergence(dts, prob, alg)
  @test sim.𝒪est[:final] ≈ OrdinaryDiffEq.alg_order(alg) atol=testTol
end
for prob in test_problems_nonlinear
  sim = test_convergence(dts, prob, alg)
  @test sim.𝒪est[:final] ≈ OrdinaryDiffEq.alg_order(alg) atol=testTol
end
integ = init(prob_ode_large, alg, adaptive=false,dt=1.e-2, save_start=false, save_end=false, save_everystep=false)
@test Base.summarysize(integ) ÷ Base.summarysize(u0_large) <= 9
integ = init(prob_ode_large, alg, adaptive=true,dt=1.e-2, save_start=false, save_end=false, save_everystep=false)
@test Base.summarysize(integ) ÷ Base.summarysize(u0_large) <= 10
integ = init(prob_ode_large, alg, dt=1.e-2, save_start=false, save_end=false, save_everystep=false, alias_u0=true)
@test Base.summarysize(integ) ÷ Base.summarysize(u0_large) <= 9
# test whether aliasing u0 is bad
new_prob_ode_nonlinear_inplace = ODEProblem(prob_ode_nonlinear_inplace.f,[1.],(0.,0.5))
sol_old = solve(prob_ode_nonlinear_inplace, alg, dt=1.e-4, save_everystep=false, save_start=false)
sol_new = solve(new_prob_ode_nonlinear_inplace, alg, dt=1.e-4, save_everystep=false, save_start=false, alias_u0=true)
@test sol_old[end] ≈ sol_new[end]
```

File: src/alg_utils.jl [hunk 1]
```diff
@@ -441,3 +447,9 @@ uses_uprev(alg::CKLLSRK54_3C, adaptive::Bool) = adaptive
 uses_uprev(alg::CKLLSRK95_4S, adaptive::Bool) = adaptive
 uses_uprev(alg::CKLLSRK95_4C, adaptive::Bool) = adaptive
 uses_uprev(alg::CKLLSRK95_4M, adaptive::Bool) = adaptive
+uses_uprev(alg::CKLLSRK54_3C_3R, adaptive::Bool) = adaptive
```
Comment (ChrisRackauckas) [inline src/alg_utils.jl]: it doesn't look like you ever use uprev?

Comment (deeepeshthakur) [reply src/alg_utils.jl]: No,I a step fails then `uprev` will be used to revert back to original position.

Comment (ChrisRackauckas) [reply src/alg_utils.jl]: oh okay yes

**Pull request merged**